### PR TITLE
CONSOLE-2535: Translation of login templates

### PR DESCRIPTION
**Pull request by sg00dwin**
relates to https://github.com/openshift/oauth-server/pull/71

### CODE REVIEW

File: _layouts/login.html [hunk 1]
```diff
@@ -10,7 +10,7 @@
         </header>
         <main class="pf-c-login__main">
```
Comment (rhamilto) [inline _layouts/login.html]:
```suggestion
        {% raw %}<main class="pf-c-login__main">
```

File: _layouts/login.html [hunk 2]
```diff
@@ -10,7 +10,7 @@
...
           <header class="pf-c-login__main-header">
-            <h1 class="pf-c-title pf-m-3xl">Log in to your account</h1>
+            <h1 class="pf-c-title pf-m-3xl">{{ .Locale.LogInToYourAccount }}</h1>
           </header>
           <div class="pf-c-login__main-body">
             {% raw %}<form class="pf-c-form" role="form" action="{{ .Action }}" method="POST">
```
Comment (rhamilto) [inline _layouts/login.html]:
```suggestion
            <form class="pf-c-form" role="form" action="{{ .Action }}" method="POST">
```

File: _layouts/login.html [hunk 3]
```diff
@@ -28,26 +28,26 @@ <h1 class="pf-c-title pf-m-3xl">Log in to your account</h1>
...
+                  <span class="pf-c-form__label-text">{{ .Locale.Password }}</span>
                   <span class="pf-c-form__label-required" aria-hidden="true">*</span>
                 </label>
                 <input type="password" class="pf-c-form-control" id="inputPassword" placeholder="" tabindex="2" type="password" name="{{ .Names.Password }}" value="">
               </div>
               <div class="pf-c-form__group pf-m-action">
-                <button class="pf-c-button pf-m-primary pf-m-block" type="submit" tabindex="3">Log in</button>
+                <button class="pf-c-button pf-m-primary pf-m-block" type="submit" tabindex="3">{{ .Locale.LogIn }}</button>
               </div>
             </form>{% endraw %}
```
Comment (rhamilto) [inline _layouts/login.html]:
```suggestion
            </form>
```

File: _layouts/login.html [hunk 4]
```diff
@@ -28,26 +28,26 @@ <h1 class="pf-c-title pf-m-3xl">Log in to your account</h1>
...
           </div>
         </main>
```
Comment (rhamilto) [inline _layouts/login.html]:
```suggestion
        </main>{% endraw %}
```

File: _layouts/login.html [hunk 5]
```diff
@@ -28,26 +28,26 @@ <h1 class="pf-c-title pf-m-3xl">Log in to your account</h1>
...
         <footer class="pf-c-login__footer">
-          <p>Welcome to {{ collection.name }}.</p>
+          <p>{{ .Locale.WelcomeTo }} {{ collection.name }}.</p>
```
Comment (rhamilto) [inline _layouts/login.html]:
```suggestion
          <p>{% raw %}{{ .Locale.WelcomeTo }}{% endraw %} {{ collection.name }}{% raw %}{{ .Locale.WelcomeToEnd }}{% endraw %}</p>
```
But this leaves the product name in english.  Not sure if we want that or want to translate it.  The logo above is in english, so maybe it's ok?

**Review by rhamilto — COMMENTED** [review]
Need to move or add `raw` tags to encompass anything that needs to be process by the server.

**Comment by rhamilto** [comment]
Also need to account for `{{page.title}}` in https://github.com/openshift/oauth-templates/blob/master/_includes/head.html#L2 .  And I notice we use ` · ` as the separator in the console's `<title>` instead of ` - `.  We should probably match the console here.

**Comment by rhamilto** [comment]
I put up a PR with these fixes.  https://github.com/sg00dwin/oauth-templates/pull/1

File: _includes/head.html [hunk 6]
```diff
@@ -1,5 +1,14 @@
 <head>
-    <title>{{page.title}} - {{ collection.name }}</title>
+    <title>
+      {% if page.layout == 'errors' %}
+        {% raw %}{{ .Locale.TitleError }}{% endraw %}
+      {% elsif page.layout == 'login' %}
+        {% raw %}{{ .Locale.TitleLogin }}{% endraw %}
+      {% elsif page.layout == 'providers' %}
+        {% raw %}{{ .Locale.TitleProviders }}{% endraw %}
```
Comment (rhamilto) [inline _includes/head.html:8]:
```suggestion
        {% raw %}{{ .Locale.TitleLogin }}{% endraw %}
```

**Comment by rhamilto** [comment]
Would be good to add the following at https://github.com/openshift/oauth-templates/blame/master/README.md#L81 since the secret is now managed by `authentication-operator`.

```
1. Scale `authentication-operator` to zero pods in the web console by visiting `http://<HOSTNAME>/k8s/ns/openshift-authentication-operator/deployments/authentication-operator` and using the pod donut controls to set pods to zero.

**Comment by rhamilto** [comment]
Whoops.  I forgot to mention we need to update the undo instructions, too.  Add the following at the bottom of README.md.

```
1. Visit `http://<HOSTNAME>/k8s/ns/openshift-authentication-operator/deployments/authentication-operator` and use the pod donut controls to set pods to one.
```

**Comment by jhadvig** [comment]
@sg00dwin ^^

**Comment by sg00dwin** [comment]
README additions added to pr

File: README.md [hunk 7]
```diff
@@ -94,3 +95,4 @@ To test the changes:
 To undo the changes for testing:
 
 1. Resume rollouts of `cluster-version-operator` in the web console by visiting `https://<HOSTNAME>/k8s/ns/openshift-cluster-version/deployments/cluster-version-operator` via `Actions > Resume Rollouts` and scale pods back to one using the pod donut controls.
+1. Visit `http://<HOSTNAME>/k8s/ns/openshift-authentication-operator/deployments/authentication-operator` and use the pod donut controls to set pods to one.
```
Comment (rhamilto) [inline README.md]: it's hard to get good help these days...
```suggestion
1. Visit `https://<HOSTNAME>/k8s/ns/openshift-authentication-operator/deployments/authentication-operator` and use the pod donut controls to set pods to one.
```

File: README.md [hunk 8]
```diff
@@ -78,6 +78,7 @@ The OCP branded templates take advantage of the override mechanism by providing
 To test the changes:
 
 1. Scale `cluster-version-operator` to zero pods in the web console by visiting `https://<HOSTNAME>/k8s/ns/openshift-cluster-version/deployments/cluster-version-operator` and using the pod donut controls to set pods to zero.  Additionally, pause rollouts via `Actions > Pause Rollouts`.
+1. Scale `authentication-operator` to zero pods in the web console by visiting `http://<HOSTNAME>/k8s/ns/openshift-authentication-operator/deployments/authentication-operator` and using the pod donut controls to set pods to zero.
```
Comment (rhamilto) [inline README.md]:
```suggestion
1. Scale `authentication-operator` to zero pods in the web console by visiting `https://<HOSTNAME>/k8s/ns/openshift-authentication-operator/deployments/authentication-operator` and using the pod donut controls to set pods to zero.
```

**Comment by rhamilto** [comment]
@sg00dwin, this ready for merge now?

**Pull request merged**